mne/stats/regression.py

```diff
@@ -134,3 +136,193 @@ def _fit_lm(data, design_matrix, names):
         mlog10_p_val[predictor] = -np.log10(p_val[predictor])
 
     return beta, stderr, t_val, p_val, mlog10_p_val
+
+
+def linear_regression_raw(raw, events, event_id=None, tmin=-.1, tmax=1,
+                          covariates=None, reject=None, tstep=1.,
+                          decim=1, picks=None, solver='pinv'):
+    """Estimate regression-based evoked potentials/fields by linear modelling
+    of the full M/EEG time course, including correction for overlapping
+    potentials and allowing for continuous/scalar predictors. Internally, this
+    constructs a predictor matrix X of size n_samples * (n_conds * window
+    length), solving the linear system Y = bX and returning b as evoked-like
+    time series split by condition.
+
+    See Smith, N. J., & Kutas, M. (2015). Regression-based estimation of ERP
+    waveforms: II. Non-linear effects, overlap correction, and practical
+    considerations. Psychophysiology, 52(2), 169-189.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        A raw object. Warning: be very careful about data that is not
+        downsampled, as the resulting matrices can be enormous and easily
+        overload your computer. Typically, 100 hz sampling rate is
+        appropriate - or using the decim keyword (see below).
+    events : np.ndarray of int, shape(n_events, 3)
+        An array where the first column corresponds to samples in raw
+        and the last to integer codes in event_id.
+    event_id : dict
+        As in Epochs; a dictionary where the values may be integers or
+        list-like collections of integers, corresponding to the 3rd column of
+        events, and the keys are condition names.
+    tmin : float | dict
+        If float, gives the lower limit (in seconds) for the time window for
+        which all event types' effects are estimated. If a dict, can be used to
+        specify time windows for specific event types: keys correspond to keys
+        in event_id and/or covariates; for missing values, the default (-.1) is
+        used.
+    tmax : float | dict
+        If float, gives the upper limit (in seconds) for the time window for
+        which all event types' effects are estimated. If a dict, can be used to
+        specify time windows for specific event types: keys correspond to keys
+        in event_id and/or covariates; for missing values, the default (1.) is
+        used.
+    covariates : dict-like | None
+        If dict-like, values have to be array-like and of the same length as
+        the columns in ```events```. Keys correspond to additional event
+        types/conditions to be estimated and are matched with the time points
+        given by the first column of ```events```. If None, only binary events
+        (from event_id) are used.
+    reject : None | dict
```

Comment:
Yes, it's very important for MEG. It may mess up the noise estimates / scalings of the inverse solution if you have flat segments / channels.